### PR TITLE
[KYUUBI #1962] Make create SparkSession as Future

### LINKED ISSUE #1962: [Improvement] Make sparkSession or SparkContext as a scala.concurrent.Future to give it proper timeout
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

https://github.com/apache/incubator-kyuubi/issues/1960 will fix a certain problem with master deploy mode, for client mode, the timeout describe there will almost not happen. The sc might get stuck will the driver is on and wait for the ApplicaitonMaster to be allocated. Although spark will have a timeout for checking for driver-am connected. But this may not happen when the AM is not able to be allocated in time.

### How should we improve?

We need a timeout for this whole progress.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by wForget**
### _Why are the changes needed?_

close #1962 

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [X] Add screenshots for manual tests if appropriate
image

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by SteNicholas** [comment]
@wForget, could you please add the unit tests for this timeout change?

**Comment by wForget** [comment]
> @wForget, could you please add the unit tests for this timeout change?

Sure, I will add later.

**Comment by ulysses-you** [comment]
@wForget can you fix the test ? hope we can land this at 1.5.0

**Comment by wForget** [comment]
> @wForget can you fix the test ? hope we can land this at 1.5.0

sure, i am doing it. 😄

**Comment by wForget** [comment]
@ulysses-you There seem to be two irrelevant failures.
```
KyuubiHiveDriverSuite:
- get tables with kyuubi driver *** FAILED ***
  java.sql.SQLException: Error operating EXECUTE_STATEMENT: org.apache.spark.sql.AnalysisException: The namespace in session catalog must have exactly one name part: hadoop_prod.default.hdp_cat_tbl;
	at org.apache.spark.sql.connector.catalog.LookupCatalog$SessionCatalogAndIdentifier$.unapply(LookupCatalog.scala:65)
	at org.apache.spark.sql.catalyst.analysis.ResolveSessionCatalog$TempViewOrV1Table$.unapply(ResolveSessionCatalog.scala:682)
	at org.apache.spark.sql.catalyst.analysis.ResolveSessionCatalog$$anonfun$apply$1.applyOrElse(ResolveSessionCatalog.scala:339)
	at org.apache.spark.sql.catalyst.analysis.ResolveSessionCatalog$$anonfun$apply$1.applyOrElse(ResolveSessionCatalog.scala:48)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.$anonfun$resolveOperatorsUp$3(AnalysisHelper.scala:90)
	at org.apache.spark.sql.catalyst.trees.CurrentOrigin$.withOrigin(TreeNode.scala:75)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.$anonfun$resolveOperatorsUp$1(AnalysisHelper.scala:90)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper$.allowInvokingTransformsInAnalyzer(AnalysisHelper.scala:212)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.resolveOperatorsUp(AnalysisHelper.scala:86)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.resolveOperatorsUp$(AnalysisHelper.scala:84)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.resolveOperatorsUp(LogicalPlan.scala:29)
	at org.apache.spark.sql.catalyst.analysis.ResolveSessionCatalog.apply(ResolveSessionCatalog.scala:48)
	at org.apache.spark.sql.catalyst.analysis.ResolveSessionCatalog.apply(ResolveSessionCatalog.scala:39)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$execute$2(RuleExecutor.scala:149)
	at scala.collection.LinearSeqOptimized.foldLeft(LinearSeqOptimized.scala:126)
	at scala.collection.LinearSeqOptimized.foldLeft$(LinearSeqOptimized.scala:122)
	at scala.collection.immutable.List.foldLeft(List.scala:91)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$execute$1(RuleExecutor.scala:146)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$execute$1$adapted(RuleExecutor.scala:138)
	at scala.collection.immutable.List.foreach(List.scala:431)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor.execute(RuleExecutor.scala:138)
	at org.apache.spark.sql.catalyst.analysis.Analyzer.org$apache$spark$sql$catalyst$analysis$Analyzer$$executeSameContext(Analyzer.scala:171)
	at org.apache.spark.sql.catalyst.analysis.Analyzer.execute(Analyzer.scala:165)
	at org.apache.spark.sql.catalyst.analysis.Analyzer.execute(Analyzer.scala:130)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$executeAndTrack$1(RuleExecutor.scala:116)
	at org.apache.spark.sql.catalyst.QueryPlanningTracker$.withTracker(QueryPlanningTracker.scala:88)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor.executeAndTrack(RuleExecutor.scala:116)
	at org.apache.spark.sql.catalyst.analysis.Analyzer.$anonfun$executeAndCheck$1(Analyzer.scala:149)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper$.markInAnalyzer(AnalysisHelper.scala:219)
	at org.apache.spark.sql.catalyst.analysis.Analyzer.executeAndCheck(Analyzer.scala:148)
	at org.apache.spark.sql.execution.QueryExecution.$anonfun$analyzed$1(QueryExecution.scala:68)
	at org.apache.spark.sql.catalyst.QueryPlanningTracker.measurePhase(QueryPlanningTracker.scala:111)
	at org.apache.spark.sql.execution.QueryExecution.$anonfun$executePhase$1(QueryExecution.scala:138)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:767)
	at org.apache.spark.sql.execution.QueryExecution.executePhase(QueryExecution.scala:138)
	at org.apache.spark.sql.execution.QueryExecution.analyzed$lzycompute(QueryExecution.scala:68)
	at org.apache.spark.sql.execution.QueryExecution.analyzed(QueryExecution.scala:66)
	at org.apache.spark.sql.execution.QueryExecution.assertAnalyzed(QueryExecution.scala:58)
	at org.apache.spark.sql.Dataset$.$anonfun$ofRows$2(Dataset.scala:99)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:767)
	at org.apache.spark.sql.Dataset$.ofRows(Dataset.scala:97)
	at org.apache.spark.sql.SparkSession.$anonfun$sql$1(SparkSession.scala:610)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:767)
	at org.apache.spark.sql.SparkSession.sql(SparkSession.scala:605)
	at org.apache.kyuubi.engine.spark.operation.ExecuteStatement.$anonfun$executeStatement$1(ExecuteStatement.scala:80)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at org.apache.kyuubi.engine.spark.operation.SparkOperation.withLocalProperties(SparkOperation.scala:88)
	at org.apache.kyuubi.engine.spark.operation.ExecuteStatement.org$apache$kyuubi$engine$spark$operation$ExecuteStatement$$executeStatement(ExecuteStatement.scala:74)
	at org.apache.kyuubi.engine.spark.operation.ExecuteStatement$$anon$1.run(ExecuteStatement.scala:106)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
  at org.apache.kyuubi.jdbc.hive.KyuubiStatement.waitForOperationToComplete(KyuubiStatement.java:400)
  at org.apache.kyuubi.jdbc.hive.KyuubiStatement.executeWithConfOverlay(KyuubiStatement.java:250)
  at org.apache.kyuubi.jdbc.hive.KyuubiStatement.execute(KyuubiStatement.java:244)
  at org.apache.kyuubi.jdbc.KyuubiHiveDriverSuite.$anonfun$new$1(KyuubiHiveDriverSuite.scala:68)
  at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
  at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
  at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
  at org.scalatest.Transformer.apply(Transformer.scala:22)
  at org.scalatest.Transformer.apply(Transformer.scala:20)
  at org.scalatest.funsuite.AnyFunSuiteLike$$anon$1.apply(AnyFunSuiteLike.scala:226)
  ...
```

```
SchedulerPoolSuite:
- Scheduler pool *** FAILED ***
  The code passed to eventually never returned normally. Attempted 201 times over 3.008023161 seconds. Last failure 
```

**Comment by ulysses-you** [comment]
it seems relevant, the attempt also failed.. cc @pan3793

**Comment by wForget** [comment]
> it seems relevant, the attempt also failed.. cc @pan3793

I see, because I do not clean up the system property.

**Comment by wForget** [comment]
```
05:21:09.634 SparkSQLSessionManager-exec-pool: Thread-764 DEBUG SparkSqlParser: Parsing command: CREATE TABLE hadoop_prod.default.hdp_cat_tbl(key int) USING iceberg
05:21:09.861 SparkSQLSessionManager-exec-pool: Thread-764 DEBUG HadoopTableOperations: Metadata for table not found in directory /home/runner/work/incubator-kyuubi/incubator-kyuubi/externals/kyuubi-spark-sql-engine/target/tmp/kyuubi-8e2e5f77-6b03-4d9a-b548-b87e6ff59c89/default/hdp_cat_tbl/metadata
java.io.FileNotFoundException: File /home/runner/work/incubator-kyuubi/incubator-kyuubi/externals/kyuubi-spark-sql-engine/target/tmp/kyuubi-8e2e5f77-6b03-4d9a-b548-b87e6ff59c89/default/hdp_cat_tbl/metadata/version-hint.text does not exist
	at org.apache.hadoop.fs.RawLocalFileSystem.deprecatedGetFileStatus(RawLocalFileSystem.java:779) ~[hadoop-client-api-3.3.2.jar:?]
	at org.apache.hadoop.fs.RawLocalFileSystem.getFileLinkStatusInternal(RawLocalFileSystem.java:1100) ~[hadoop-client-api-3.3.2.jar:?]
	at org.apache.hadoop.fs.RawLocalFileSystem.getFileStatus(RawLocalFileSystem.java:769) ~[hadoop-client-api-3.3.2.jar:?]
	at org.apache.hadoop.fs.FilterFileSystem.getFileStatus(FilterFileSystem.java:462) ~[hadoop-client-api-3.3.2.jar:?]
	at org.apache.hadoop.fs.ChecksumFileSystem$ChecksumFSInputChecker.<init>(ChecksumFileSystem.java:160) ~[hadoop-client-api-3.3.2.jar:?]
	at org.apache.hadoop.fs.ChecksumFileSystem.open(ChecksumFileSystem.java:372) ~[hadoop-client-api-3.3.2.jar:?]
	at org.apache.hadoop.fs.FileSystem.open(FileSystem.java:976) ~[hadoop-client-api-3.3.2.jar:?]
	at org.apache.iceberg.hadoop.HadoopTableOperations.findVersion(HadoopTableOperations.java:309) ~[iceberg-spark-runtime-3.1_2.12-0.13.1.jar:?]
	at org.apache.iceberg.hadoop.HadoopTableOperations.refresh(HadoopTableOperations.java:102) ~[iceberg-spark-runtime-3.1_2.12-0.13.1.jar:?]
	at org.apache.iceberg.hadoop.HadoopTableOperations.current(HadoopTableOperations.java:83) ~[iceberg-spark-runtime-3.1_2.12-0.13.1.jar:?]
	at org.apache.iceberg.BaseMetastoreCatalog.loadTable(BaseMetastoreCatalog.java:42) ~[iceberg-spark-runtime-3.1_2.12-0.13.1.jar:?]
	at org.apache.iceberg.shaded.com.github.benmanes.caffeine.cache.BoundedLocalCache.lambda$doComputeIfAbsent$14(BoundedLocalCache.java:2344) ~[iceberg-spark-runtime-3.1_2.12-0.13.1.jar:?]
	at java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1908) ~[?:?]
	at org.apache.iceberg.shaded.com.github.benmanes.caffeine.cache.BoundedLocalCache.doComputeIfAbsent(BoundedLocalCache.java:2342) ~[iceberg-spark-runtime-3.1_2.12-0.13.1.jar:?]
	at org.apache.iceberg.shaded.com.github.benmanes.caffeine.cache.BoundedLocalCache.computeIfAbsent(BoundedLocalCache.java:2325) ~[iceberg-spark-runtime-3.1_2.12-0.13.1.jar:?]
	at org.apache.iceberg.shaded.com.github.benmanes.caffeine.cache.LocalCache.computeIfAbsent(LocalCache.java:108) ~[iceberg-spark-runtime-3.1_2.12-0.13.1.jar:?]
	at org.apache.iceberg.shaded.com.github.benmanes.caffeine.cache.LocalManualCache.get(LocalManualCache.java:62) ~[iceberg-spark-runtime-3.1_2.12-0.13.1.jar:?]
	at org.apache.iceberg.CachingCatalog.loadTable(CachingCatalog.java:161) ~[iceberg-spark-runtime-3.1_2.12-0.13.1.jar:?]
	at org.apache.iceberg.spark.SparkCatalog.load(SparkCatalog.java:488) ~[iceberg-spark-runtime-3.1_2.12-0.13.1.jar:?]
	at org.apache.iceberg.spark.SparkCatalog.loadTable(SparkCatalog.java:135) ~[iceberg-spark-runtime-3.1_2.12-0.13.1.jar:?]
	at org.apache.iceberg.spark.SparkCatalog.loadTable(SparkCatalog.java:92) ~[iceberg-spark-runtime-3.1_2.12-0.13.1.jar:?]
	at org.apache.spark.sql.connector.catalog.TableCatalog.tableExists(TableCatalog.java:119) ~[spark-catalyst_2.12-3.1.3.jar:3.1.3]
	at org.apache.spark.sql.execution.datasources.v2.CreateTableExec.run(CreateTableExec.scala:39) ~[spark-sql_2.12-3.1.3.jar:3.1.3]
	at org.apache.spark.sql.execution.datasources.v2.V2CommandExec.result$lzycompute(V2CommandExec.scala:40) ~[spark-sql_2.12-3.1.3.jar:3.1.3]
	at org.apache.spark.sql.execution.datasources.v2.V2CommandExec.result(V2CommandExec.scala:40) ~[spark-sql_2.12-3.1.3.jar:3.1.3]
	at org.apache.spark.sql.execution.datasources.v2.V2CommandExec.executeCollect(V2CommandExec.scala:46) ~[spark-sql_2.12-3.1.3.jar:3.1.3]
	at org.apache.spark.sql.Dataset.$anonfun$logicalPlan$1(Dataset.scala:228) ~[spark-sql_2.12-3.1.3.jar:3.1.3]
	at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:3700) ~[spark-sql_2.12-3.1.3.jar:3.1.3]
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$5(SQLExecution.scala:103) ~[spark-sql_2.12-3.1.3.jar:3.1.3]
	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:163) ~[spark-sql_2.12-3.1.3.jar:3.1.3]
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:90) ~[spark-sql_2.12-3.1.3.jar:3.1.3]
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:775) ~[spark-sql_2.12-3.1.3.jar:3.1.3]
	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64) ~[spark-sql_2.12-3.1.3.jar:3.1.3]
	at org.apache.spark.sql.Dataset.withAction(Dataset.scala:3698) ~[spark-sql_2.12-3.1.3.jar:3.1.3]
	at org.apache.spark.sql.Dataset.<init>(Dataset.scala:228) ~[spark-sql_2.12-3.1.3.jar:3.1.3]
	at org.apache.spark.sql.Dataset$.$anonfun$ofRows$2(Dataset.scala:99) ~[spark-sql_2.12-3.1.3.jar:3.1.3]
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:775) ~[spark-sql_2.12-3.1.3.jar:3.1.3]
	at org.apache.spark.sql.Dataset$.ofRows(Dataset.scala:96) ~[spark-sql_2.12-3.1.3.jar:3.1.3]
	at org.apache.spark.sql.SparkSession.$anonfun$sql$1(SparkSession.scala:618) ~[spark-sql_2.12-3.1.3.jar:3.1.3]
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:775) ~[spark-sql_2.12-3.1.3.jar:3.1.3]
	at org.apache.spark.sql.SparkSession.sql(SparkSession.scala:613) ~[spark-sql_2.12-3.1.3.jar:3.1.3]
	at org.apache.kyuubi.engine.spark.operation.ExecuteStatement.$anonfun$executeStatement$1(ExecuteStatement.scala:80) ~[classes/:?]
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23) ~[scala-library-2.12.15.jar:?]
	at org.apache.kyuubi.engine.spark.operation.SparkOperation.withLocalProperties(SparkOperation.scala:88) ~[classes/:?]
	at org.apache.kyuubi.engine.spark.operation.ExecuteStatement.org$apache$kyuubi$engine$spark$operation$ExecuteStatement$$executeStatement(ExecuteStatement.scala:74) ~[classes/:?]
	at org.apache.kyuubi.engine.spark.operation.ExecuteStatement$$anon$1.run(ExecuteStatement.scala:106) ~[classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	at java.lang.Thread.run(Thread.java:829) ~[?:?]
```

**Comment by wForget** [comment]
@ulysses-you @pan3793 Is this failure related? i have no idea about this.
```
IcebergOperationSuite:
- get catalogs *** FAILED ***
  "[spark_catalog]" did not equal "[hadoop_prod]" (IcebergMetadataTests.scala:32)
  Analysis:
  "[spark_catalog]" -> "[hadoop_prod]"
- get schemas *** FAILED ***
  List("db1", "db2", "db33", "db44", "global_temp") did not contain "default" (HiveJDBCTestHelper.scala:92)
- get schemas with multipart namespaces *** FAILED ***
  List("`a.b`", "`a.b`.c", "a.`b.c`", "`a.b.c`", "`a.b``.c`", "db1.db2.db3", "db4", "global_temp", "a", "db1", "db1.db2") did not contain "default" (HiveJDBCTestHelper.scala:92)
```

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala [hunk 1]
```diff
@@ -20,6 +20,9 @@ package org.apache.kyuubi.engine.spark
 import java.time.Instant
 import java.util.concurrent.CountDownLatch
 
+import scala.concurrent.{Await, Future, TimeoutException}
+import scala.concurrent.ExecutionContext.Implicits.global
```
Comment (pan3793) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala:24]: You should use a dedicated `ExecutionContext` to start Spark

File: kyuubi-common/src/test/scala/org/apache/kyuubi/KyuubiFunSuite.scala [hunk 2]
```diff
@@ -134,6 +134,23 @@ trait KyuubiFunSuite extends AnyFunSuite
     }
   }
 
+  final def withSystemProperty(properties: Map[String, String])(f: => Unit): Unit = {
+    val originValues = properties.map {
+      case (key, value) =>
+        val originValue = System.getProperty(value)
```
Comment (pan3793) [inline kyuubi-common/src/test/scala/org/apache/kyuubi/KyuubiFunSuite.scala:140]: `System.getProperty(value)` value? do you mean key?

Comment (wForget) [reply kyuubi-common/src/test/scala/org/apache/kyuubi/KyuubiFunSuite.scala:140]: > `System.getProperty(value)` value? do you mean key?

yeah, i wiil fix it

**Comment by wForget** [comment]
@ulysses-you could you help me rerun checks?

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala [hunk 3]
```diff
@@ -83,6 +85,9 @@ object SparkSQLEngine extends Logging {
 
   private val countDownLatch = new CountDownLatch(1)
 
+  implicit private val ec: ExecutionContext =
+    ExecutionContext.fromExecutor(Executors.newFixedThreadPool(1))
```
Comment (pan3793) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala:89]: we need shutdown this executor, otherwire will cause thread leak

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala [hunk 4]
```diff
@@ -196,7 +201,11 @@ object SparkSQLEngine extends Logging {
     } else {
       var spark: SparkSession = null
       try {
-        spark = createSpark()
+        val sparkFuture = Future {
```
Comment (pan3793) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala:204]: we can create the ec here and shutdown in finally clause

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala (hunk 3 above)
Comment (pan3793) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala:89]: And it doesn't make sense for all creations to share one thread

Comment (wForget) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala:89]: In my opinion, for different engines, the createpark method will be executed in different processes and only once. 
Is there a problem with this understanding?

Comment (pan3793) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala:89]: You are right, the last comment is wrong, ignore it.

**Comment by wForget** [comment]
This failure seems to be related to the JDK version. the `HiveMetaStoreClient` is not properly cleaned up.

`org.apache.hadoop.hive.metastore.HiveMetaStore.HMSHandler#threadLocalMS#get()` returns the same object in multiple test cases.

**Comment by ulysses-you** [comment]
It seems dangerous if we initialize spark and use hive client in different thread. we can initialize spark in current thread and cancel it or not according to timeout in a new thread.

**Comment by wForget** [comment]
> It seems dangerous if we initialize spark and use hive client in different thread. 

@ulysses-you For kyuubi, it always seems to work like this, `createSpark` and `session operator` are not always the same thread.

Execute init sqls after future , the test case seems to execute fine, but I'm not sure if that's ok.

**Comment by wForget** [comment]
reimplemented in #2230, close this.

**Pull request closed without merging**